.scripts/create_upgrade.php

```diff
@@ -75,14 +66,49 @@
 
 $upgrade_code = <<<___UPGRADE
 <?php
+
+namespace Elgg\Upgrades;
+
 /**
- * Elgg $release upgrade $upgrade_version
- * $name
+ * Elgg $release upgrade $class_name
  *
- * Description
+ * Add description here
  */
+class $class_name implements Upgrade {
+
+	public function isRequired() {
+		// Check the database or the datadir to check if there is something
+		// that this upgrade needs to process. Return true/false depending
+		// on the result.
+	}
+
+	public function getTitle() {
+		return 'Add user facing title here';
+	}
+
+	public function getDescription() {
+		return 'Add user facing description here';
+	}
+
+	public function run() {
+		// Do the actual upgrading here
+	}
+
+	public function getTimeCreated() {
+		// Do not modify this manually
+		return $time;
+	}
 
-// upgrade code here.
+	public function getRelease() {
+		// Do not modify this manually
+		return '$upgrade_release';
+	}
+
+	public function getUniqueName() {
+		// Do not modify this manually
+		return '{$time}-{$upgrade_release}-{$class_name}';
```

Comment:
We need a unique name to store in the list of successfully ran upgrades.

What do you think of getting it like this? I wouldn't want to make the name of the class similar monster as the old upgrades (e.g. 2015031300-1.11.0_dev-comment-access-sync-50c9764e5845315c.php).

I'm not  sure if it's a good idea to include important features like this into a class that the plugin dev is supposed to edit.